tests/suites/test_suite_ssl.function

```diff
@@ -198,7 +198,7 @@ void mbedtls_test_message_queue_free( mbedtls_test_message_queue *queue )
  *
  * \retval  MBEDTLS_TEST_ERROR_ARG_NULL, if the queue is null.
  * \retval  MBEDTLS_TEST_ERROR_QUEUE_FULL, if the queue is full.
- * \retval  0, if the push was succesfull.
+ * \retval  \p len, if the push was successful.
```

Comment:
Consider applying "Change the recv and send callbacks API to return the length of data" as a fixup to "Add a message metadata queue in ssl tests"

Reply:
I introduced the previous structure of commits to show how some of the changes are justified - for example, a silent fail upon message truncation, which is concise with the existing API, but I'm not comfortable with. Changed that though, according to the suggestions.